components/style/counter_style/mod.rs

```diff
@@ -444,7 +444,16 @@ impl Parse for Negative {
 ///
 /// Empty Vec represents 'auto'
 #[derive(Clone, Debug)]
-pub struct Ranges(pub Vec<Range<Option<i32>>>);
+pub struct Ranges(pub Vec<Range<CounterBound>>);
+
+/// A bound found in `Ranges`.
+#[derive(Clone, Copy, Debug)]
+pub enum CounterBound {
+    /// An integer bound.
+    Integer(i32),
```

Comment:
This should be an integer and support `calc`, mind filing an issue?

Reply:
I filed https://github.com/servo/servo/issues/20197.